scripts/src/packagemapping/generatelocks.py

```diff
@@ -0,0 +1,106 @@
+from glob import glob
+import sys
+import yaml
+import re
+from datetime import datetime, timezone
+from json import dumps as to_json
+
+
+def ownerfile_regex():
+    """Return a regex statement that parses relative filepaths of an OWNERS file.
+
+    The filepath is expected to be from the charts directory forward.
+
+    Returns:
+        The compiled regex that groups the category, organization, and chart name.
+
+        E.g. category | organization | chartname
+             partner  | hashicorp    | vault
+    """
+
+    pattern = re.compile(r"charts/(partners|redhat|community)/([\w-]+)/([\w-]+)/OWNERS")
+    return pattern
+
+
+def logError(msg, file=sys.stderr):
+    """logError just prints the msg with an ERROR caption to stderr unless otherwise defined"""
+    print(f"[ERROR] {msg}", file=file)
+
+
+def logInfo(msg, file=sys.stderr):
+    """logError just prints the msg with an INFO caption to stderr unless otherwise defined"""
+    print(f"[INFO] {msg}", file=file)
+
+
+def logWarn(msg, file=sys.stderr):
+    """logWarn just prints the msg with an INFO caption to stderr unless otherwise defined"""
+    print(f"[WARN] {msg}", file=file)
+
+
+def main():
+    packages = {}
+    for filename in glob("charts/**/OWNERS", recursive=True):
+        logInfo(f"processing file {filename}")
+        pattern = ownerfile_regex()
+        matched = pattern.match(filename)
+        if matched is None:
+            # TODO(komish): Consider if we need to have a flag that exits and otherwise make this a warning
```

Comment:
Definitely -

When this script was first written, it had to combat multiple chart directories that had invalid names. There were no charts associated with these, presumably, because of the incorrect naming scheme. A good example is a chart that has a name starting with a bracket, e.g. `[`. This is invalid in multiple places; I'm pretty sure a chart cannot start with the `[` character.

The original intention here was to allow for those to exist in repo but still not cause workflows that ran this script to fail, such that a lock would not be accounted for when parsing the failing entry.

In the grand scheme, however, I think this comment can and should be removed. We pretty much always want this to fail in cases where the input is invalid. I'll remove these lines in the next push.